src/Workspaces/Core/Portable/Shared/Utilities/IWorkspaceThreadingService.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.Shared.Utilities
+{
+    /// <summary>
+    /// An optional interface which allows an environment to customize the behavior for synchronous methods that need to
+    /// block on the result of an asynchronous invocation. An implementation of this is provided in the MEF catalog when
+    /// applicable.
```

Comment:
I will admit this description felt really pithy; if it just said "gives an abstraction for JTF.Run" that would have almost immediately made sense.

Reply:
I think I'd also like to see us just call this something more like the "global options threading service" or something more specific to options, because I don't expect (or want) any new code calling this. Any new code should just be natively async.

Reply:
> I think I'd also like to see us just call this something more like the "global options threading service" or something more specific to options, because I don't expect (or want) any new code calling this.

It's not intended to be specific to options, or specific to JTF. It's a general helper for cases where an application dictates a specific way to invoke an asynchronous method and block on the result. For VS, the specific way is JTF.

Reply:
The documentation is now more complete